pkg/network/ebpf/c/protocols/postgres/helpers.h

```diff
@@ -19,6 +19,20 @@ static __always_inline bool is_postgres_connect(const char *buf, __u32 buf_size)
     return !bpf_memcmp(buf + sizeof(*hdr), PG_STARTUP_USER_PARAM, sizeof(PG_STARTUP_USER_PARAM));
 }
 
+// Classify ping query of postgres.
+static __always_inline bool is_ping(const char *buf, __u32 buf_size) {
+    if (buf_size < sizeof(POSTGRES_PING_BODY)) {
+        return false;
+    }
+    char tmp[sizeof(POSTGRES_PING_BODY)] = {0};
+    // Cannot use bpf_memcpy here because of verifier error of "misaligned stack access".
+#pragma unroll (sizeof(POSTGRES_PING_BODY))
+    for (int i = 0; i < sizeof(POSTGRES_PING_BODY); i++) {
+        tmp[i] = buf[i];
+    }
```

Comment:
are we fine with leaving it like that? I mean maybe we should investigate the root case for that as it pretty wired?

Reply:
it is not weird, the original buffer is allocated on the stack, and we are trying access from odd location (index 5 and on)
the verifier reject that behavior